src/borg/platform/linux.pyx

```diff
@@ -136,17 +138,31 @@ def set_flags(path, bsd_flags, fd=None):
         if stat.S_ISBLK(st.st_mode) or stat.S_ISCHR(st.st_mode) or stat.S_ISLNK(st.st_mode):
             # See comment in get_flags().
             return
-    cdef int flags = 0
+    cdef int flags
+    cdef int mask = LINUX_MASK  # 1 at positions we want to influence
+    cdef int new_flags = 0
     for bsd_flag, linux_flag in BSD_TO_LINUX_FLAGS.items():
         if bsd_flags & bsd_flag:
-            flags |= linux_flag
+            new_flags |= linux_flag
+
     open_fd = fd is None
     if open_fd:
         fd = os.open(path, os.O_RDONLY|os.O_NONBLOCK|os.O_NOFOLLOW)
     try:
+        # Get current flags. If this fails, fall back to 0 so we can still attempt to set.
+        if ioctl(fd, FS_IOC_GETFLAGS, &flags) == -1:
+            flags = 0
```

Comment:
Discuss: shall we use flags=0 and try to continue or shall we rather just give up and not try to set the flags if we can't determine the current flags from the fs?